pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -243,7 +260,17 @@ func shutdownAppsGlobal(ctx *zedagentContext) {
 var baseOSPrevConfigHash []byte
 
 func parseBaseOS(getconfigCtx *getconfigContext,
-	config *zconfig.EdgeDevConfig) {
+	config *zconfig.EdgeDevConfig) (activateNewBaseOSFlag bool) {
+	activateNewBaseOSFlag = false
+
+	// check whether any baseOS config is switched to the Activate state
+	for _, baseOS := range getconfigCtx.pubBaseOsConfig.GetAll() {
+		baseOSConfig := baseOS.(types.BaseOsConfig)
+		if baseOSConfig.Activate {
+			activateNewBaseOSFlag = true
```

Comment:
But this doesn't detect a change. And normally there is one BaseOsConfig with the activate flag set so in practice this will almost always be true.

Reply:
Fixed